plugins/ua/universalarchive.plugin.zsh

```diff
@@ -0,0 +1,59 @@
+alias ua='universalarchive'
+alias ax='universalarchive tar.xz '
+
+# exclusively POSIX shell syntax is used
+
+universalarchive() { # compress a file or folder
+  local file
+  local parent
+  local name
+
+  local type="$1"
+  local input="$2"
+
+  if [ -f "$input" ]; then
+    if [ "$#" -eq 2 ]; then
+      name="${input%.[^.]*}"
+    else
+      file="$(realpath "$input")" 
+      parent="${file%/*}" 
+      name="${parent##*/}"
+    fi
+  elif [ -d "$input" ]; then
+    name="${input%/*}"
+  else
+    echo "universalarchive(): archive a file or directory."
+    echo "Usage:   ua <archive type> <files>"
+    echo "Error: Please give a valid file or directory name!"
+  fi
+
+  if [ -f "${name}.${type}" ]; then
+    name=$(mktemp --dry-run "./${name}_XXX.${type}")
+  fi
+
+  case "$type" in
+    tar)          shift; tar      cvf            "${name}.tar"       "${@%%}" ;;
```

Comment:
If you're going to run `shift` in all cases you can put it outside the `case` switch.

Reply:
Thank you. In case `*` shift is not run, but I put it outside anyway.